src/api/spec/factories/project.rb

```diff
@@ -11,18 +11,14 @@
 
     factory :project_with_package do
       transient do
-        package_name nil
+        sequence(:package_name) { |n| "package_#{n}" }
         create_patchinfo false
         maintainer nil
       end
 
       after(:create) do |project, evaluator|
-        new_package = if evaluator.package_name
-                        create(:package, project_id: project.id, name: evaluator.package_name)
-                      else
-                        create(:package, project_id: project.id)
-                      end
-        project.packages << new_package
+        package = Package.where(project_id: project.id, name: evaluator.package_name).first
+        project.packages << (package || create(:package, project_id: project.id, name: evaluator.package_name))
```

Comment:
Beside I don't see the point of the sequence by default and the search of the package to avoid duplicates...
If you want to replace the old **if**... It would be better to do something like this (letting the package_name to nil by default)?:
```ruby
project.packages << create(:package, { project_id: project.id, name: evaluator.package_name }.delete_if { |k, v| v.nil? } )
```